test/python/tests/replica/features/pool.feature

```diff
@@ -18,7 +18,7 @@ Feature: Mayastor pool management
   Scenario: creating a pool with a name that already exists
     Given a pool "p0"
     When the user creates a pool with the name of an existing pool
-    Then the pool creation should succeed
+    Then the pool create command should fail
```

Comment:
Should we have this change in old API also, because consumers of old API does not expect pool creation to fail if pool already exists?

Reply:
We are taking these in develop and we are eventually moving towards changing the control plane according. Taking this will help us find any control plane changes needed early.

Reply:
Fail is a bit generic, it should specify the error code/type, otherwise it will be difficult to determine if it's a failure creating the pool (let's say device error) or if it's because it already exists.
Will there be a documentation for the semantic changes?
Seems that now Pool's have a uuid as well? should it also be checked if it already exists?

Reply:
For create if it exists we return a `EEXIST` and rest of the checks will return `EINVAL`. And yes there should be documentation for it. I don't have one but will come up with one as soon as possible.

For the uuid, the create request does not have the uuid field. The import rpc is now separate and has a optional uuid field that will check for the uuid if provided.

Reply:
Would it be possible to add the uuid for the create?

Reply:
IMO for the create request uuid should not be needed as we won't have a uuid with us before the pool is created. For import the uuid can be helpful as we know the uuid of an exported pool we need to import.

Reply:
hmm, so what happens if the create times out, but ends up succeeding. Now the control plane doesn't know the uuid of the pool, so how does it reconcile things?

Reply:
As we added a filter for the name in ListPools, we can check for the pool before creating it.
The reconcile loop would be like
```
  pool = ListPools(ListPoolOptions(name="tpool"))
  if pool == nil {
      CreatePool(name="tpool", disks=["/dev/sdc"])
  }

```

Reply:
right but then we're reverting to just checking by name and not uuid. So in that case the control plane could instead just set the name to a uuid (assuming the name can be that long) and keep its own name<->uuid mapping.

Reply:
Can there be multiple pools with the same name? If no how can uuid help with reconciling? Why do we need to keep a different name mapping in the control plane?

Reply:
I don't know if it can or not. If it can then the uuid would be the tie breaker if you will. Let's say you plug a pool in the mayastor node which happens to have the same name as another pool, how does the control plane know which one is which without a uuid? Ofcourse the uuid can also be the same, but in that case nothing will help :D

Reply:
Ohhh now it make sense, it is better to have it and not use it than to not have it :sweat_smile: . On the same lines can you take a look at the proto file for the messages https://github.com/openebs/mayastor-api/pull/8/files

Reply:
So for the Create request the uuid will be optional and it will be checked only if provided. And we still need to fail if the pool exists for the given name or uuid. right?

Reply:
Yes both lvm and lvs have a uuid, not in the same format but unique in the way spdk and lvm generates one.

Reply:
Sure we can add a check to lookup_by_uuid and reject the request if one already exists.

Reply:
FYI: For LVM pools and volumes the uuid will always be auto generated, we can never set it manually.

Reply:
Yep @shubham14bajpai explained this to me offline. We may have a solution for LVM adding the uuid as a tag, and this way hopefully be able to keep the same uuid format for different pool types.